### PR TITLE
Unit test setup carbon graphs package

### PR DESCRIPTION
**Pull request by AshishMotanamGurunadham**
### Summary

Closes # None

- This is dummy PR created to test scripts in Travis CI.
### Deployment Link

https://terra-graphs-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,12 +20,15 @@ jobs:
         - npm run danger
         - npm run lint
         - npm run jest
+        - npm run unit
         # These can't be built in parallel because the cache is a single object and you can't add to it in parallel
         - npm run compile:prod -- --env.disableAggregateThemes --output-path $TRAVIS_BUILD_DIR/travis-build/default
         - npm run compile:lowlight -- --output-path $TRAVIS_BUILD_DIR/travis-build/lowlight
         - npm run compile:fusion -- --output-path $TRAVIS_BUILD_DIR/travis-build/fusion
+    - stage: build
```
Comment (emilyrohrbough) [inline .travis.yml:28]: For the build stage to run, you'll wana add to the stages config below; https://github.com/cerner/terra-graphs/pull/4/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485R169

File: README.md [hunk 2]
```diff
@@ -14,7 +14,30 @@
...
+[![Build Status](https://travis-ci.com/cerner/carbon-graphs.svg?branch=main)](https://travis-ci.com/cerner/carbon-graphs)
+[![npm latest version](https://img.shields.io/npm/v/@cerner/carbon-graphs/latest.svg)](https://www.npmjs.com/package/@cerner/carbon-graphs)
+[![Cerner OSS](https://badgen.net/badge/Cerner/OSS/blue)](https://engineering.cerner.com/open_source/)
+[![License](https://badgen.net/badge/license/Apache-2.0/blue)](https://github.com/cerner/carbon-graphs/blob/main/LICENSE)
+[![dependencies Status](https://david-dm.org/cerner/carbon-graphs/status.svg)](https://david-dm.org/cerner/carbon-graphs)
+[![devDependencies Status](https://david-dm.org/cerner/carbon-graphs/dev-status.svg)](https://david-dm.org/cerner/carbon-graphs?type=dev)
+[![Bugs](https://img.shields.io/github/issues/cerner/carbon-graphs/bug.svg)](https://github.com/cerner/carbon-graphs/issues?utf8=✓&q=is%3Aissue+is%3Aopen+label%3Abug)
+[![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
+
+
```
Comment (emilyrohrbough) [inline README.md:29]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -26,6 +49,107 @@
 - [Local Development](#local-development)
 - [LICENSE](#license)
 
+
+
```
Comment (emilyrohrbough) [inline README.md:53]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -26,6 +49,107 @@
...
+  - [Deprecated](#deprecated)
+- [Supported Browsers](#supported-browsers)
+- [Internationalization (I18n)](#internationalization-i18n)
+  - [Packages Requiring I18n](#packages-requiring-i18n)
+- [Contributing](#contributing)
+- [Local Development](#local-development)
+- [LICENSE](#license) -->
+
+<hr>
+<!--  -->
```
Comment (emilyrohrbough) [inline README.md:67]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -26,6 +49,107 @@
...
+| Browser                     | Version |
+|-----------------------------|---------|
+| Chrome & Chrome for Android | Current |
+| Edge                        | Current |
+| Firefox                     | Current |
+| Internet Explorer           | 10 & 11 |
+| Safari & Mobile Safari      | Current |
+
+<hr>
+<!--  -->
```
Comment (emilyrohrbough) [inline README.md:80]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -26,6 +49,107 @@
...
+npm i @cerner/Terra Graphs-graphs --save-dev
+```
+
+### Getting Started
+
+-   [Getting Started](./docs/getting-started/GettingStarted.md)
+-   [API Reference](docs/README.md)
+
+<hr>
+<!--  -->
```
Comment (emilyrohrbough) [inline README.md:106]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -26,6 +49,107 @@
...
+-   Supports custom SVG shapes for data-sets
+-   `Non-contiguous` data point line graph
+-   `Sparkline` graph (with Shapes as hidden)
+-   Criticality indicators for data point
+-   Locale support for axes ticks
+-   Y and Y2 Axes label along with respective shapes
+-   Graph Panning
+
+<hr>
+<!--  -->
```
Comment (emilyrohrbough) [inline README.md:151]:
```suggestion
```

File: docs/contributing/README.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+        -   [Release](#release)
+        -   [Deploying the site to GitHub](#deploying-the-site-to-github)
+    -   [Breaking changes](#breaking-changes)
+
+This project uses NPM for dependency management and provides NPM scripts to call on `Webpack` to run tasks on the project.
+To get started, you will need to checkout this project.
+
+## Code of Conduct
+
+Carbon adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).
```
Comment (emilyrohrbough) [inline docs/contributing/README.md:25]: Will we align this code of conduct with Terra's?

https://github.com/cerner/terra-core/blob/main/CODE_OF_CONDUCT.md

File: docs/contributing/README.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+We try to make process as simple as possible:
+
+![Alt](../assets/contribution-model.png "Model")
+
+Before starting development on a Carbon project, please complete the following steps:
+
+1. If contribution involves changes to `requirements`, create your work in the Carbon JIRA project using the appropriate template:
+    - <a href="https://jira3.cerner.com/browse/CARBON-1/">Enhancement</a>
+    - <a href="https://jira3.cerner.com/browse/CARBON-2/">Defect</a>
```
Comment (emilyrohrbough) [inline docs/contributing/README.md:41]: Remove JIRA reference & create a .git template for these issue types?

examples: https://github.com/cerner/terra-core/tree/main/.github/ISSUE_TEMPLATE

File: docs/contributing/README.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+We follow `ESLint` [commit convention](https://eslint.org/docs/1.0.0/developer-guide/contributing#step-2-make-your-changes).
+Your commit message should look something like this:
+
+```text
+<Tag>: <Short description> (fixes #1234)
+```
+
+**Note**:
+1234 is the Carbon [Github issue](https://github.com/cerner/carbon-graphs/issues/237) number.
```
Comment (emilyrohrbough) [inline docs/contributing/README.md:103]:
```suggestion
**Note**:
1234 in the above example iis the Carbon [Github issue](https://github.com/cerner/carbon-graphs/issues/1234) number.
```

File: docs/contributing/README.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+carbon-graphs will be released following release types based on the commit message tags
+
+|   Tag    | Release |
+| :------: | :-----: |
+| Breaking |  major  |
+|   Fix    |  patch  |
+|   New    |  minor  |
+|  Update  |  minor  |
+| Upgrade  |  patch  |
```
Comment (emilyrohrbough) [inline docs/contributing/README.md:125]: What is the difference between update and upgrade?

File: docs/controls/Bubble.md [hunk 12]
```diff
@@ -0,0 +1,342 @@
+# Bubble
+
+**Note: Bubble will be deprecated in a future release. Use [BubbleSingleDataset](./BubbleSingleDataset.md) or [BubbleMultipleDataset](./BubbleMultipleDataset.md).**
```
Comment (emilyrohrbough) [inline docs/controls/Bubble.md:3]: If it's gonna be deprecated, does it need to be added here?

File: docs/controls/BubbleMultipleDataset.md [hunk 13]
```diff
@@ -0,0 +1,283 @@
...
+            -   [Required](#required-2)
+            -   [Optional](#optional-2)
+        -   [Regions](#regions)
+            -   [Required](#required-3)
+            -   [Optional](#optional-3)
+        -   [Constraints](#constraints)
+
+## Usage
+
+**Note: Bubble Muliple Dataset is limited to 7 datasets. any additional datasets will be ignored.**
```
Comment (emilyrohrbough) [inline docs/controls/BubbleMultipleDataset.md:26]:
```suggestion
**Note: Bubble Multiple Dataset is limited to 7 datasets. any additional datasets will be ignored.**
```

File: package.json [hunk 14]
```diff
@@ -46,6 +46,35 @@
           "import/first": "off",
           "import/no-duplicates": "off"
         }
+      },
+      {
+        "files": [
+          "*.js"
+        ],
+        "rules": {
+          "import/no-cycle": "off",
```
Comment (emilyrohrbough) [inline package.json:55]: Should this warn? Ideally these would be resolved

File: package.json [hunk 15]
```diff
@@ -124,22 +124,68 @@
...
+    "url-loader": "^4.1.1",
     "webpack": "^4.30.0",
     "webpack-cli": "^3.3.2",
     "webpack-dev-server": "^3.3.1"
+  },
+  "scripts": {
+    "build": "webpack --config ./packages/carbon-graphs/webpack/webpack.config.js --env.TYPE=LIB",
+    "build:dist": "webpack --config ./packages/carbon-graphs/webpack/webpack.config.js --env.TYPE=DIST",
+    "bootstrap:hoist": "lerna bootstrap --hoist",
+    "clean": "lerna clean --yes && rm -rf node_modules || true",
```
Comment (emilyrohrbough) [inline package.json:154]:
```suggestion
    "clean": "lerna clean --yes; rm -rf node_modules",
```

File: package.json [hunk 16]
```diff
@@ -124,22 +124,68 @@
...
+    "clean:install": "npm run clean && npm install",
```
Comment (emilyrohrbough) [inline package.json:155]:
```suggestion
    "clean:install": "npm run clean; npm install",
```

File: packages/carbon-graphs/LICENSE [hunk 17]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright <%= currentYear %> Cerner Innovation, Inc.
```
Comment (emilyrohrbough) [inline packages/carbon-graphs/LICENSE:190]: Add year

**Review by emilyrohrbough — COMMENTED** [review]
Was each piece of carbon-graphs indented to be released as it's own package or will they all be in one package named terra-graphs?

If they are all going to be in one package, what other modules do we expect to be added to this mono-repo?

File: packages/terra-graphs/package.json [hunk 18]
```diff
@@ -18,22 +15,114 @@
...
+    "cssnano": "^4.1.7",
+    "docdash": "^1.2.0",
+    "eslint": "^6.8.0",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-config-standard": "^14.1.1",
+    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-jsdoc": "^24.0.6",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-promise": "^4.2.1",
+    "eslint-plugin-standard": "^4.0.1",
```
Comment (emilyrohrbough) [inline packages/terra-graphs/package.json:89]: These should have been defined at the root level already and can be removed here. Bootstrap will pull these up for package-level testing

File: packages/terra-graphs/package.json [hunk 19]
```diff
@@ -18,22 +15,114 @@
...
+    "@cerner/svg-to-carbon": "^1.1.3",
+    "@commitlint/cli": "^8.3.5",
+    "@commitlint/config-conventional": "^8.3.4",
+    "@semantic-release/git": "^9.0.0",
+    "autoprefixer": "^9.8.4",
+    "babel-loader": "^8.1.0",
+    "babel-plugin-istanbul": "^6.0.0",
+    "babel-plugin-minify-replace": "^0.5.0",
+    "babel-preset-minify": "^0.5.1",
+    "babel-eslint": "^10.1.0",
```
Comment (emilyrohrbough) [inline packages/terra-graphs/package.json:76]: Same comment as the eslint comment below

File: packages/terra-graphs/package.json [hunk 20]
```diff
@@ -18,22 +15,114 @@
...
+    "terra-theme-context": "^1.0.0",
+    "d3": "^5.16.0"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.10.4",
+    "@babel/plugin-transform-classes": "^7.10.4",
+    "@babel/plugin-transform-runtime": "^7.10.4",
+    "@babel/polyfill": "^7.10.4",
+    "@babel/preset-env": "^7.10.4",
+    "@babel/runtime": "^7.10.4",
```
Comment (emilyrohrbough) [inline packages/terra-graphs/package.json:65]: Same comment as the eslint comment below for these babel

File: packages/terra-graphs/package.json [hunk 21]
```diff
@@ -18,22 +15,114 @@
...
+    "karma-webpack": "^4.0.2",
+    "less": "^3.11.2",
+    "less-loader": "^6.2.0",
+    "loader-utils": "^1.4.0",
+    "mini-css-extract-plugin": "^0.9.0",
+    "one-cerner-style-icons": "github:cerner/one-cerner-style-icons",
+    "postcss-loader": "^3.0.0",
+    "prettier": "^2.0.5",
+    "puppeteer": "^3.3.0",
+    "rimraf": "^3.0.2",
```
Comment (emilyrohrbough) [inline packages/terra-graphs/package.json:111]:
```suggestion
```

File: packages/terra-graphs/package.json [hunk 22]
```diff
@@ -18,22 +15,114 @@
...
+    "semantic-release": "^17.1.1",
+    "sinon": "^9.0.2",
+    "style-loader": "^1.2.1",
+    "stylelint": "^13.6.1",
```
Comment (emilyrohrbough) [inline packages/terra-graphs/package.json:115]:
```suggestion
```

File: packages/terra-graphs/package.json [hunk 23]
```diff
@@ -18,22 +15,114 @@
...
+    "stylelint-config-standard": "^20.0.0",
```
Comment (emilyrohrbough) [inline packages/terra-graphs/package.json:116]: I'd maybe push this to the root level to follow the same stylelint standard in the mono repo. It's currently setup to lint for terra's standard: https://github.com/cerner/terra-graphs/blob/main/package.json#L63

File: packages/terra-graphs/package.json [hunk 24]
```diff
@@ -18,22 +15,114 @@
...
+    "@commitlint/config-conventional": "^8.3.4",
+    "@semantic-release/git": "^9.0.0",
+    "autoprefixer": "^9.8.4",
+    "babel-loader": "^8.1.0",
+    "babel-plugin-istanbul": "^6.0.0",
+    "babel-plugin-minify-replace": "^0.5.0",
+    "babel-preset-minify": "^0.5.1",
+    "babel-eslint": "^10.1.0",
+    "commander": "^5.1.0",
+    "conventional-changelog-eslint": "^3.0.8",
```
Comment (emilyrohrbough) [inline packages/terra-graphs/package.json:78]: I'd look at pulling to the root package.json level

File: packages/carbon-graphs/webpack/site.webpack.js [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+    ],
+  },
+  plugins: [
+    new MiniCssExtractPlugin({
+      filename: 'carbon-graphs.css',
+      chunkFilename: 'carbon-graphs.css',
+    }),
+    new HtmlWebpackPlugin({
+      // packages/carbon-graphs/dev-site/assets/index.ejs
+      // ../assets/index.ejs
```
Comment (emilyrohrbough) [inline packages/carbon-graphs/webpack/site.webpack.js:47]:
```suggestion
```

File: packages/carbon-graphs/webpack/site.webpack.js [hunk 26]
```diff
@@ -0,0 +1,69 @@
+const path = require('path');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const { jsOptions, cssOptions } = require('./helpers/rules');
+
+const webpackConfig = (env) => ({
+  mode: 'development',
+  entry: ['./packages/carbon-graphs/dev/app.js'],
+  output: {
+    path: path.resolve(__dirname, '../../..', '.site'),
```
Comment (emilyrohrbough) [inline packages/carbon-graphs/webpack/site.webpack.js:10]:
```suggestion
    path: path.resolve(__dirname, '..', '..' , '..', '.site'),
```

File: packages/carbon-graphs/webpack/site.webpack.js [hunk 27]
```diff
@@ -0,0 +1,69 @@
...
+      },
+      {
+        test: /\.(less|css)$/,
+        use: [MiniCssExtractPlugin.loader, ...cssOptions(env.TYPE)],
+      },
+      {
+        test: /\.(js|jsx)$/,
+        include: [
+
+          path.join(__dirname, '../src/main/js'),
```
Comment (emilyrohrbough) [inline packages/carbon-graphs/webpack/site.webpack.js:33]:
```suggestion
          path.join(__dirname, '..', 'src', 'main', 'js'),
```

File: docs/controls/Bubble.md (hunk 12 above)
Comment (sdadn) [reply docs/controls/Bubble.md:3]: We don't know when the next major release will be, so we added this note here for current consumers.

File: docs/controls/BubbleMultipleDataset.md (hunk 13 above)
Comment (sdadn) [reply docs/controls/BubbleMultipleDataset.md:26]: Fixed here: 82579d2ee67fe882d14efe63eb9ad5ed09c2629d

File: packages/carbon-graphs/LICENSE (hunk 17 above)
Comment (sdadn) [reply packages/carbon-graphs/LICENSE:190]: added: 44268284731bb22e6931c1aa16d835edd1341074

File: packages/terra-graphs/package.json (hunk 18 above)
Comment (sdadn) [reply packages/terra-graphs/package.json:89]: Yup, they have been moved up.

File: packages/terra-graphs/package.json (hunk 21 above)
Comment (sdadn) [reply packages/terra-graphs/package.json:111]: cbb350dcef98535345b5be3a92435d1baa5a59a0

File: packages/terra-graphs/package.json (hunk 22 above)
Comment (sdadn) [reply packages/terra-graphs/package.json:115]: cbb350dcef98535345b5be3a92435d1baa5a59a0

File: packages/terra-graphs/package.json (hunk 23 above)
Comment (sdadn) [reply packages/terra-graphs/package.json:116]: yeah, it's at root now.

File: packages/carbon-graphs/webpack/site.webpack.js (hunk 25 above)
Comment (sdadn) [reply packages/carbon-graphs/webpack/site.webpack.js:47]: 25e1c97439f1ae40aaabe046ab8c24152507e271

File: package.json (hunk 14 above)
Comment (sdadn) [reply package.json:55]: The linter rules will be reviewed after the migration:
https://github.com/cerner/terra-graphs/issues/8

**Pull request closed without merging**